cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -1907,14 +1916,14 @@ protected String getUrl(String path) {
     }
 
     @SuppressWarnings("unchecked")
-    private String addPageOfResources(String nextUrl, List<Map<String, Object>> allResources) {
+    private String addPageOfResources(String nextUrl, List<Map<String, Object>> allResources, String next) {
         String resp = getRestTemplate().getForObject(getUrl(nextUrl), String.class);
         Map<String, Object> respMap = JsonUtil.convertJsonToMap(resp);
         List<Map<String, Object>> newResources = (List<Map<String, Object>>) respMap.get("resources");
         if (newResources != null && newResources.size() > 0) {
             allResources.addAll(newResources);
         }
-        return (String) respMap.get("next_url");
+        return (String) respMap.get(next);
```

Comment:
But isn't the `next` field nested inside a `pagination` field? How would this work for V3 then?

Reply:
Refactored